### PR TITLE
Serial communication Protocol to Arduino working with base Stepper Position.

### PR DESCRIPTION
**Pull request by lamwx**

### CODE REVIEW

File: ArduinoCode/ArduinoCode.ino [hunk 1]
```diff
@@ -186,47 +214,64 @@ void UpdateGripper() {
 }
 
 double RotateFeedback() {
-  // have specific feedback for the rotation joint here
-  return 0.0;
+//rotates negative until it hits the limit switch, then roates to 0 and sets zero
+//had some werid behaviour trying to set zero at -180.
+  double current;
```
Comment (newmanjoel) [inline ArduinoCode/ArduinoCode.ino:219]: Should this be set to static? Just so its not created new each time?

File: ArduinoCode/ArduinoCode.ino [hunk 2]
```diff
@@ -186,47 +214,64 @@ void UpdateGripper() {
...
 double GripperFeedback() {
   // have specific feedback for the gripper joint here
   return 0.0;
 }
-void OutputRotate(){
-  // have specific output for the rotate joint here
+void OutputRotate() {
+  int steps;
+  steps = RotateSetpoint * stepsPerRevolution / 360 - rotateCurrentSteps;
+  rotateCurrentSteps = rotateCurrentSteps + steps;
```
Comment (newmanjoel) [inline ArduinoCode/ArduinoCode.ino:262]: Should this not be done in the feedback?

**Review by newmanjoel — APPROVED** [review]
I like this. Only thing I would do is maybe rearrange some of where the stepper motor feedback comes from. Other than that its great! Thanks!

**Comment by newmanjoel** [comment]
#2

**Comment by lamwx** [comment]
Sounds good, I’ll rearange those. And yep that should definitely be a static.

**Pull request merged**